src/celeritas/optical/ScintillationParams.hh

```diff
@@ -27,22 +28,37 @@ class ScintillationParams final : public ParamsDataInterface<ScintillationData>
   public:
     //!@{
     //! \name Type aliases
+    using SPConstParticles = std::shared_ptr<ParticleParams const>;
     using ScintillationDataCRef = HostCRef<ScintillationData>;
     //!@}
 
-    //! Material-dependent scintillation data, indexed by \c OpticalMaterialId
+    //! Scintillation data for all materials
     struct Input
     {
-        std::vector<ImportScintData> data;
+        using VecOptMatId = std::vector<OpticalMaterialId>;
+        using VecSPId = std::vector<ScintillationParticleId>;
+
+        VecOptMatId matid_to_optmatid;  //!< MaterialId to OpticalMaterialId
+        VecSPId pid_to_scintpid;  //!< ParticleId to ScintillationParticleId
+        std::vector<ImportScintData> data;  //!< Indexed by OpticalMaterialId
+        bool scintillation_by_particle;  //!< Particle or material sampling
```

Comment:
isn't this redundant with whether matid_to_optmatid or pid_to_scintpid is not empty? It's also missing a default initializer.

Reply:
Not really, since `matid_to_optmatid` and `pid_to_scintpid` are always the size of the total material list and total particle list. So even if we sample only by material, we still know which particles are being associated to the scintillation process. Of course, if the boolean is true and we do sample scintillation by particle, then we'd also use the `pid_to_scintpid` along with `ScintillationData::spectrum_index(...)` to get the index of `ScintillationData::particles` that is storing said specific particle+material spectrum.